model/task/task.go

```diff
@@ -2470,83 +2471,103 @@ func (t *Task) Insert() error {
 // into the old_tasks collection. If this is a display task, its execution tasks
 // are also archived.
 func (t *Task) Archive() error {
-	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
-		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
-		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+	if t.DisplayOnly {
+		var task []Task
+		task = append(task, *t)
+		err := ArchiveMany(task)
+		return errors.Wrapf(err, "archiving and updating display task with ID '%s'.", t.Id)
+	} else {
+		// Archiving a single task.
+		if !t.ResetFailedWhenFinished || evergreen.IsFailedTaskStatus(t.Status) {
```

Comment:
I think when we consider ResetFailed we should also consider ResetWhenFinished, maybe making a helper function or etc (although I still think this isn't the right place to consider it I'm not sure where the right place is for this comment). it's possible that we set ResetFailedWhenFinished bc an execution task was stranded for example, but a user chose to restart the whole thing, i.e. ResetWhenFinished was set. This should take precedence.